src/RepoDriver.sol

```diff
@@ -0,0 +1,530 @@
+    /// including the case of each letter and special characters being removed.
+    /// @return userId The ID of the user.
+    function requestUpdateOwner(Forge forge, bytes memory name)
+        public
+        whenNotPaused
+        returns (uint256 userId)
+    {
+        uint256 fee = _repoDriverAnyApiStorage().defaultFee;
+        require(linkToken.balanceOf(address(this)) >= fee, "Link balance too low");
+        return _requestUpdateOwner(forge, name, fee);
+    }
+
+    /// @notice The function called when receiving funds from ERC-677 `transferAndCall`.
+    /// Only supports receiving Link tokens, callable only by the Link token smart contract.
+    /// The only supported usage is requesting user ownership updates,
+    /// the transferred tokens are then used for paying the AnyApi operator fee,
+    /// see `requestUpdateOwner` for more details.
+    /// The received tokens are never refunded, so make sure that
+    /// the amount isn't too low to cover the fee, isn't too high and wasteful,
+    /// and the repository's content is valid so its ownership can be verified.
+    /// @param amount The transferred amount, it will be used as the AnyApi operator fee.
+    /// @param data The `transferAndCall` payload.
+    /// It must be a valid ABI-encoded calldata for `requestUpdateOwner`.
+    /// The call parameters will be used the same way as when calling `requestUpdateOwner`,
+    /// to determine which user's ownership update is requested.
+    function onTokenTransfer(address, /* sender */ uint256 amount, bytes calldata data)
+        public
+        whenNotPaused
+    {
+        require(msg.sender == address(linkToken), "Callable only by the Link token");
+        require(data.length >= 4, "Data not a valid calldata");
+        require(bytes4(data[:4]) == this.requestUpdateOwner.selector, "Data not requestUpdateOwner");
+        (Forge forge, bytes memory name) = abi.decode(data[4:], (Forge, bytes));
+        _requestUpdateOwner(forge, name, amount);
+    }
+
+    /// @notice Requests an update of the ownership of the user representing the repository.
+    /// See `requestUpdateOwner` for more details.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @param fee The fee in Link to pay for the request.
+    /// @return userId The ID of the user.
+    function _requestUpdateOwner(Forge forge, bytes memory name, uint256 fee)
+        internal
+        returns (uint256 userId)
+    {
+        RepoDriverAnyApiStorage storage storageRef = _repoDriverAnyApiStorage();
+        address operator = address(storageRef.operator);
+        require(operator != address(0), "Operator address not set");
+        uint256 nonce = storageRef.nonce++;
+        bytes32 requestId = keccak256(abi.encodePacked(this, nonce));
+        userId = calcUserId(forge, name);
+        storageRef.requestedUpdates[requestId] = userId;
```

Comment:
Wondering if we should track the pending requests with a counter. Which is increased here and decreased after request fulfillment.

Reply:
A counter like `mapping(uint256 userId => uint256) numOfPendingRequests`? What uses would it have?

Reply:
I thought more of a global one. :thinking: